feat(TAD-6145): init Linechart component

cbaune-talend opened the pull request:
**What is the problem this PR is trying to solve?**
Currently the dataviz package does not include any linegraph

**What is the chosen solution to this problem?**
Add a linegraph component in the dataviz package

**Please check if the PR fulfills these requirements**

- [x] The PR have used `yarn changeset` to a request a release from the CI if wanted.
- [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
- [x] Tests for the changes have been added (for bug fixes / features) And [non reg](./screenshots.md) done before need review
- [x] Docs have been added / updated (for bug fixes / features)
- [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

<!-- You can add more checkboxes here -->

**[ ] This PR introduces a breaking change**

<!-- if the PR introduces a breaking change, add the description here. So when you merge this PR, add this description into the [breaking change wiki](https://github.com/Talend/ui/wiki/BREAKING-CHANGE) in the next version -->

<!-- **Original Template** -->

<!-- https://github.com/Talend/tools/blob/master/tools-root-github/.github/PULL_REQUEST_TEMPLATE.md -->

Nigui commented on packages/dataviz/src/components/LineChart/LineChart.component.tsx:
```diff
@@ -0,0 +1,79 @@
+			data={data}
+			margin={margin || {}}
+		  >
+			{ !!showGridLines && <CartesianGrid stroke="#F2F2F2" strokeDasharray="2" vertical={false} />}
```
Could use one of those [gray values](https://talend.surge.sh/design-system/?path=/docs/design-tokens-design-tokens--page&globals=theme:light#grayscale) or use directly design token

Benrajalu replied:
Those gray values are indeed deprecated. Sorry, we'll remove that part of the doc soon. The tokens are the source of truth: https://talend.surge.sh/design-tokens/?path=/docs/welcome--page

Nigui commented on packages/dataviz/src/components/LineChart/LineChart.types.ts:
```diff
@@ -0,0 +1,45 @@
+	}
+	tooltip?: {
+		custom?: boolean,
+		constentStyle?: any,
```
typing error ? `contentStyle` ?

Nigui commented on packages/dataviz/src/components/LineChart/LineChartTooltip.component.tsx:
```diff
@@ -0,0 +1,59 @@
+	const getLineUnit = (axis: 'left' | 'right' | undefined) => axis === 'right' ? chartOptions.rightYAxisOptions?.rechartsOptions?.unit : chartOptions.leftYAxisOptions?.rechartsOptions?.unit ;
+
+	const getLineIconBackground = (strokeDasharray: string | number | undefined, color: string) => {
+		console.log('stroke : ', strokeDasharray);
```
to remove

Nigui reviewed (commented):
Good start imo !
You could use design tokens for size and colors

gjou-tlnd commented on packages/dataviz/src/components/LineChart/LineChartTooltip.component.tsx:
```diff
@@ -0,0 +1,59 @@
+		<div className={classNames(styles['line-chart-custom-tooltip-wrapper'])}>
+			<p className={classNames(styles['line-chart-custom-tooltip-wrapper-title'])}>{label}</p>
+			{linesConfig.map(config => (
+		  		<p id={`tooltip_item_${config.key}`}>
+					<div
```
using div inside a p is not allowed

cbaune-talend commented:
Added some tests and more storybook examples.
Updated CSS to use design tokens.

So this is the first version of the feature.
A second version should comme after to simplify the API : taking some hard choices on the graph look&feel to greatly simplify the props of the component.

gjou-tlnd commented on packages/dataviz/src/components/LineChart/LineChart.scss:
```diff
@@ -0,0 +1,72 @@
+@use '~@talend/design-tokens/lib/_tokens.scss' as tokens;
+
+.line-chart-custom-tooltip-wrapper {
```
please use sass & BEM in this module

gjou-tlnd commented on packages/dataviz/src/components/LineChart/LineChartTooltip.component.tsx:
```diff
@@ -0,0 +1,59 @@
+						style={{ background: getLineIconBackground(config.rechartsOptions?.strokeDasharray, config.color)}}
+					/>
+					<span className={classNames(styles['line-chart-custom-tooltip-line-label'])}>
+						{ config.tooltipLabel ? config.tooltipLabel : config.key }
```
```suggestion
						{ config.tooltipLabel ?? config.key }
```

gjou-tlnd commented on packages/dataviz/src/components/LineChart/LineChartTooltip.component.tsx:
```diff
@@ -0,0 +1,59 @@
+		return color;
+	};
+
+	if (active && payload && payload.length) {
```
```suggestion
	if (active && payload?.length) {
```

gjou-tlnd commented on packages/dataviz/src/components/LineChart/LineChartTooltip.component.tsx:
```diff
@@ -0,0 +1,59 @@
+
+	const getLineUnit = (axis: 'left' | 'right' | undefined) => axis === 'right' ? chartOptions.rightYAxisOptions?.rechartsOptions?.unit : chartOptions.leftYAxisOptions?.rechartsOptions?.unit ;
+
+	const getLineIconBackground = (strokeDasharray: string | number | undefined, color: string) => {
+		if(strokeDasharray) {
+			return `repeating-linear-gradient(to right, ${color} 0, ${color} 10px,transparent 10px,transparent 12px)`;
+		}
+		return color;
+	};
```
this should be outside the component

gjou-tlnd commented on packages/dataviz/src/components/LineChart/LineChartLegend.component.tsx:
```diff
@@ -0,0 +1,79 @@
+
+	};
+
+	if (payload && payload.length) {
```
```suggestion
	if (payload?.length) {
```

Benrajalu commented on packages/dataviz/src/components/LineChart/LineChart.scss:
```diff
@@ -1,72 +1,79 @@
+		&__line-value {
+			font: tokens.$coral-paragraph-m-bold;
+			color: tokens.$coral-color-neutral-text;
+			margin-left: tokens.$coral-size-xxs;
```
```suggestion
			margin-left: tokens.$coral-spacing-xxs;
```

Benrajalu commented on packages/dataviz/src/components/LineChart/LineChart.scss:
```diff
@@ -1,72 +1,79 @@
+		flex-direction: column;
+		justify-content: center;
+		align-items: flex-start;
+		padding: tokens.$coral-size-m;
```
```suggestion
		padding: tokens.$coral-spacing-m;
```

Benrajalu commented on packages/dataviz/src/components/LineChart/LineChart.scss:
```diff
@@ -1,72 +1,79 @@
+		flex-direction: row;
+		justify-content: flex-start;
+		align-items: center;
+		gap: tokens.$coral-size-m;
```
```suggestion
		gap: tokens.$coral-spacing-m;
```

jmfrancois reviewed (changes requested):
please remove the package-lock

gjou-tlnd commented on packages/dataviz/src/components/LineChart/LineChart.component.test.tsx:
```diff
@@ -0,0 +1,192 @@
+/* eslint-disable no-console */
+import React from 'react';
+import { mount } from 'enzyme';
```
IMHO, new code should use RTL, any thoughts @jsomsanith ?

gjou-tlnd commented on packages/dataviz/src/components/LineChart/LineChart.component.test.tsx:
```diff
@@ -0,0 +1,192 @@
+/* eslint-disable no-console */
```
```suggestion
```

gjou-tlnd commented on packages/dataviz/src/components/LineChart/LineChart.stories.tsx:
```diff
@@ -0,0 +1,317 @@
+import { action } from '@storybook/addon-actions';
+import  LineChart, { LineChartProps } from './LineChart.component';
+
+const Template: Story<LineChartProps> = args => <LineChart {...args} />;
```
```suggestion
```

you can use the new CSF syntax:
```suggestion
export const FullyCustomisedLineChart = Template.bind({});```
```suggestion
const Template: Story<LineChartProps> = args => <LineChart {...args} />;
```

```
export const FullyCustomisedLineChart = {
   args: {...}
};
```

gjou-tlnd commented:
Tooltip style doesn't match bar/geo charts, is this expected? Can't we use only one style?
![Screenshot 2022-04-11 at 14 18 07](https://user-images.githubusercontent.com/58977230/162737703-92d31a9c-efec-46dc-9555-6421ce154bc2.png)
![Screenshot 2022-04-11 at 14 18 26](https://user-images.githubusercontent.com/58977230/162737768-fce111b0-f3db-428e-99d9-13306b56f79d.png)

gjou-tlnd commented:
> Tooltip style doesn't match bar/geo charts, is this expected? Can't we use only one style? ![Screenshot 2022-04-11 at 14 18 07](https://user-images.githubusercontent.com/58977230/162737703-92d31a9c-efec-46dc-9555-6421ce154bc2.png) ![Screenshot 2022-04-11 at 14 18 26](https://user-images.githubusercontent.com/58977230/162737768-fce111b0-f3db-428e-99d9-13306b56f79d.png)



> > Tooltip style doesn't match bar/geo charts, is this expected? Can't we use only one style?
> 
> If we agree on the white tooltip for all chart, we can go for it here 👍🏼

I'll let @cbaune-talend lead the subject with product designers to make sure everyone is on board

cbaune-talend commented:
Last update ! Taking into account the comments from @gjou-tlnd & @Benrajalu : 
- reworked the tooltip to stick to DS principles
- switched to RTL for testing
- use new CSF syntax for stories

gjou-tlnd commented:
> Last update ! Taking into account the comments from @gjou-tlnd & @Benrajalu :
> 
> * reworked the tooltip to stick to DS principles
> * switched to RTL for testing
> * use new CSF syntax for stories
> 
> ![image](https://user-images.githubusercontent.com/81923357/164425418-40bdebbd-1504-441e-8b43-7bbcfa46b3bc.png)

We still end up with two tooltip implementations, please merge them

Nigui reviewed (changes requested):
Please export LineChart components and types from dataviz module

Nigui commented:
> Please export LineChart components and types from dataviz module

You can apply this patch to do it 
(replace `.txt` by `.patch` before calling `> git am 0001-feat-TDC-6145-export-LineChart-component-and-types.patch` command)

[0001-feat-TDC-6145-export-LineChart-component-and-types.patch](https://github.com/Talend/ui/files/8539584/0001-feat-TDC-6145-export-LineChart-component-and-types.txt)

gjou-tlnd commented:
> @Nigui added your commit to export the component.
> 
> @gjou-tlnd : since the two tooltip component have different needs, I don't think it would be wise to merge them in a single tooltip component. None the less, I modified the `TooltipContent` css to use the same DS tokens, so the two looks the same. If we really want to make a unique tooltip component for all the graph we could create another ticket and discuss a feasible future solution.

I was thinking of the creating a container only responsible for color/border/padding and using children for actual content... but using DS tokens is a good first step, thks

jmfrancois reviewed (approved):
animation of the LineChart seems really slow compare to other charts.

cbaune-talend commented:
> animation of the LineChart seems really slow compare to other charts.

fixed in [3fb5955](https://github.com/Talend/ui/pull/3785/commits/3fb595578aae24782bab2564c1b85fb2967532c7)
